### ISSUE TITLE
List what "you" already have that could be added to BIDS-matlab 

### ISSUE THREAD

**Issue by Remi-Gau** [bids-standard/bids-matlab#61]
As discussed in the last meeting (see #59) we will be keeping track of what "people" have already created to deal BIDS that could be added to this repository

- FieldTrip (i.e. the toolbox at large) includes low-level code to read all file formats that BIDS deals with 
- [bids-matlab-tools](https://github.com/sccn/bids-matlab-tools) mentioned in this [post](https://github.com/bids-standard/bids-matlab/issues/59#issuecomment-704735183)
  - https://github.com/sccn/bids-matlab-tools/blob/master/rename_brainvision_files.m
  - https://github.com/sccn/bids-matlab-tools/blob/master/bids_spreadsheet2participants.m
- [CPP-BIDS](https://github.com/cpp-lln-lab/CPP_BIDS): to help with creating certain files (tsv and json) for raw BIDS

**Comment by Remi-Gau** [comment]
From Tibor:
Automatic analysis:
- https://github.com/automaticanalysis/automaticanalysis/blob/master/aa_engine/aas_processBIDS.m
- https://github.com/automaticanalysis/automaticanalysis/blob/master/aa_toolbox/aa_export_toBIDS.m

**Comment by Remi-Gau** [comment]
- SPM also has some BIDS functionality: https://en.wikibooks.org/wiki/SPM/BIDS
- AnyWave has BIDS tools under development: https://meg.univ-amu.fr/wiki/AnyWave:Faq#BIDS_support
- Swartz Center Matlab Tools for EEG BIDS: https://github.com/sccn/bids-matlab-tools

**Comment by bmdeen** [comment]
I'm developing a BIDS-compliant pipeline for individual subject fMRI analysis, fMRIPermPipe (fppipe-v2 branch). I've written various low-level functions for interacting with BIDS filenames and metadata files, which might be useful: https://github.com/bmdeen/fmriPermPipe/tree/fppipe-v2/%2Bfpp/%2Bbids .
Functions that may be of interest: 1) fpp.bids.changeName (modifies a BIDS-formatted filename), 2) fpp.bids.jsonChangeValue (changes value in json file), 3) fpp.bids.jsonReorient (modifies metadata for a change in MRI image orientation).

I've also written a number of wrappers for FSL functions like flirt, fnirt, applywarp, invwarp, and convert_xfm, which generate .json files with updated metadata: https://github.com/bmdeen/fmriPermPipe/tree/fppipe-v2/%2Bfpp/%2Bfsl . Probably doesn't make sense to incorporate these directly into bids-matlab, but they may be useful to others.

**Comment by Remi-Gau** [comment]
> Functions that may be of interest: 1) fpp.bids.changeName (modifies a BIDS-formatted filename), 2) fpp.bids.jsonChangeValue (changes value in json file), 3) fpp.bids.jsonReorient (modifies metadata for a change in MRI image orientation).
> 
> I've also written a number of wrappers for FSL functions like flirt, fnirt, applywarp, invwarp, and convert_xfm, which generate .json files with updated metadata: https://github.com/bmdeen/fmriPermPipe/tree/fppipe-v2/%2Bfpp/%2Bfsl . Probably doesn't make sense to incorporate these directly into bids-matlab, but they may be useful to others.

I think some of those functions could be interesting though I suspect that we might want to try to make them more general by providing bids-matlab the ability to create filenames and JSON files that are compliant with BIDS derivatives specs.

**Comment by bmdeen** [comment]
Yeah, these specific functions are not exactly what you'd want in bids-matlab - e.g., they're designed to be flexible and work with non-BIDS-compliant input filenames. I shared them anyway, since there might be some useful functionality.

**Comment by Remi-Gau** [comment]
Pieter Vandemaele: https://github.com/GIfMI/matbids

Henk Mutsaerts:https://github.com/ExploreASL/ExploreASL

**Comment by guiomar** [comment]
Hi! 
You already mentioned Fieldtrip toolbox and I saw @robertoostenveld as a contributor, so he might have included the functions already, but just in case here you can have some examples and the use of the 'data2bids' function in Fieldtrip:
https://www.fieldtriptoolbox.org/example/bids/
